src/http/ngx_http_mruby_module.c

```diff
@@ -2019,6 +2023,81 @@ static int ngx_http_mruby_set_der_certificate(ngx_ssl_conn_t *ssl_conn, ngx_str_
   return NGX_ERROR;
 }
 
+static int ngx_http_mruby_set_set_client_ca_cert_data(ngx_ssl_conn_t *ssl_conn, ngx_str_t *cert)
+{
+  BIO *bio = NULL;
+  X509 *x509 = NULL;
+  u_long n;
+
+  /* clear existing CA list name */ 
+  SSL_set_client_CA_list(ssl_conn, NULL);
+
+  /* read certificate data from memory buffer */
+  if ((bio = BIO_new_mem_buf(cert->data, cert->len)) == NULL) {
+    goto NGX_MRUBY_SSL_ERROR;
+  }
+
+  /* read rest of the chain */
+  while (!BIO_eof(bio)) {
+    x509 = PEM_read_bio_X509(bio, NULL, NULL, NULL);
+    if (x509 == NULL) {
+      n = ERR_peek_last_error();
+
+      if (ERR_GET_LIB(n) == ERR_LIB_PEM && ERR_GET_REASON(n) == PEM_R_NO_START_LINE) {
+        ERR_clear_error();
+        break;
+      }
+
+      goto NGX_MRUBY_SSL_ERROR;
+    }
+    // Add CA certificate name list
+    if (SSL_add_client_CA(ssl_conn, x509) == 0) {
+      goto NGX_MRUBY_SSL_ERROR;
+    }
```

Comment:
I couldn't find a good way to set up certificate from memory. As far I understand `SSL_CTX_load_verify_locations` method only accept disk written certificate files. So, I think we might need to write a certificate from memory to some temp file. Or I should not allow set certificate from memory.
Or if you have a better idea please suggest to me.